scipy/stats/_stats_py.py

```diff
@@ -2819,8 +2819,10 @@ def sem(a, axis=0, ddof=1, nan_policy='propagate'):
     1.2893796958227628
 
     """
-    n = a.shape[axis]
-    s = np.std(a, axis=axis, ddof=ddof) / np.sqrt(n)
+    xp = array_namespace(a)
+    a = atleast_nd(a, ndim=1, xp=xp)
+    n = xp.asarray(a.shape[axis], dtype=a.dtype)
+    s = xp.std(a, axis=axis, correction=ddof) / xp.sqrt(n)
```

Comment:
This has come up a few times, and I've done the same thing. I wonder if the preferred style should be:
```suggestion
    n = a.shape[axis]
    s = xp.std(a, axis=axis, correction=ddof) / n**0.5
```
for simplicity, since `n` is a scalar?

Reply:
or `math.sqrt`

Reply:
Yeah I was thinking that, too, if we have things that are definitely scalar. In the past, I haven't liked using `math` next to `np` (even if there's less overhead), but it makes some sense if these other libraries don't accept Python scalars.